rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistStorageProviderImplTest.java

```diff
@@ -0,0 +1,176 @@
+
+        whitelistStorageProvider.save(activationConfig);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        assertEquals(1, actualLockWhitelist.getAll().size());
+
+        //Making sure the saved value is correct
+        Map<Address, OneOffWhiteListEntry> whitelistedAddresses = getAddressFromOneOffWhiteListEntryInMemoryStorage();
+        assertTrue(whitelistedAddresses.containsKey(firstBtcAddress));
+    }
+
+    @Test
+    void save_whenIsActiveRSKIP87_ShouldSavedUnlimitedLockWhitelist() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(true);
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        //make sure the lockWhitelist is empty
+        assertEquals(0, lockWhitelist.getAll().size());
+
+        LockWhitelistEntry oneOffWhiteListEntry =  addOneOffWhiteListEntry();
+        LockWhitelistEntry unlimitedWhiteListEntry = addUnlimitedWhiteListEntry();
+        lockWhitelist.put(firstBtcAddress, oneOffWhiteListEntry);
+        lockWhitelist.put(secondBtcAddress, unlimitedWhiteListEntry);
+
+        whitelistStorageProvider.save(activationConfig);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        assertEquals(2, actualLockWhitelist.getAll().size());
+
+        //Making sure the saved value is correct
+        Map<Address, OneOffWhiteListEntry> oneOffWhiteListEntryMap = getAddressFromOneOffWhiteListEntryInMemoryStorage();
+        assertTrue(oneOffWhiteListEntryMap.containsKey(firstBtcAddress));
+
+        Map<Address, LockWhitelistEntry> lockWhitelistEntryMap = getAddressFromUnlimitedWhiteListEntryInMemoryStorage();
+        assertTrue(lockWhitelistEntryMap.containsKey(secondBtcAddress));
+    }
+
+    @Test
+    void getLockWhitelist_whenLockWhitelistIsNull_shouldReturnZeroEntries() {
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        assertEquals(0, actualLockWhitelist.getAll().size());
+    }
+
+    @Test
+    void getLockWhitelist_whenLockWhitelistIsNotNull_shouldReturnOneEntry() {
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        LockWhitelistEntry oneOffWhiteListEntry =  addOneOffWhiteListEntry();
+        lockWhitelist.put(firstBtcAddress, oneOffWhiteListEntry);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        assertEquals(1, actualLockWhitelist.getAll().size());
+        //Making sure the correct value was whitelisted
+        assertTrue(actualLockWhitelist.isWhitelisted(firstBtcAddress));
+    }
+
+    @Test
+    void getWhitelist_whenIsActiveRSKIP87_shouldReturnTwoEntries() {
```

Comment:
I don't think this is right. RSKIP87 active doesn't mean it should return 2 elements. See the implementation of the method

Reply:
I was taking a look at the method, and first, it is looking for the `LOCK_ONE_OFF` key, and after that, in `if (activations.isActive(RSKIP87))`, it is looking for the `LOCK_UNLIMITED` key. If both exist in storage, they are added to the map that later is returned. This scenario might happen. 
WDYT? Would you like an additional test or modify this?

Reply:
The scenario you describe is correct but I think the test name is confusing. It implies that after RSKIP87 is active 2 elements will be returned. That is not really the case, it could still return 1, 2,3, n elements.

`getWhitelist_whenIsActiveRSKIP87_shouldReturnUnlimitedEntries`

Something like that makes more sense to me

Reply:
Done! the method was renamed.